### PR TITLE
Preserve Data

### PR DESCRIPTION
**Pull request by momo-87**
### Added Features:
- [x] Create people.json file
- [x] Refactor `create_a_person` method to preserve data into `people.json `file
- [x] Refactor `list_all_people `method to read data from `people.json` file

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -3,23 +3,76 @@
...
+    File.foreach('people.json') do |line|
+      next if line.strip.empty?
+
+      person_data = JSON.parse(line)
+      person_class = person_data['class'] == 'Student' ? Student : Teacher
+      person = person_class.new(person_data['age'], person_data['name'])
+      person.parent_permission = false if person_class == Student && person_data['parent_permission'] == false
+      @people << person
+    end
+  end
```
Comment (souad988) [inline app.rb]: Very well done by reading the file and generating new Student and Teacher instances from the preserved data 🥇 👍🏼 , One small issue to fixe is handling the case where the json file doesn't exist to insure that the app doesn't crache and the same goes for load_books and load_rentals

File: app.rb [hunk 2]
```diff
@@ -32,49 +85,75 @@ def create_a_person
...
+    person.parent_permission = false if ans == 'N'
+    @people << person
+    save_person_to_file(person)
+  end
+
+  def create_teacher(age, name, specialization)
+    person = Teacher.new(age, specialization, name)
+    @people << person
+    save_person_to_file(person)
+  end
```
Comment (souad988) [inline app.rb]: Good job the new added person is getting stored in the file 👍🏼 👍🏼 👍🏼 , but from this projects requirements this should be done once on app exit Kindly update the save to file to be saving data from @books and @rentals and @people on app exit 

**Project requirements :**
**On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)**

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @momo-87 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


242722065-db428b0c-8dfc-49c4-a156-42f227e32868

## Highlights
Loading preserved data on load
saving data to json files

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @momo-87 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark: All requested changes by reviewer implemented :+1: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**